superset/dashboards/commands/delete.py

```diff
@@ -40,7 +41,7 @@ def __init__(self, user: User, model_id: int):
         self._model_id = model_id
         self._model: Optional[Dashboard] = None
 
-    def run(self):
+    def run(self) -> Optional[Model]:
```

Comment:
`BaseDAO.delete` seems to always return `Model` (this should be added to the signature in `superset/dao/base/BaseDAO.delete`), ie. this should also do the same?